cartographer/io/pcd_writing_points_processor.cc

```diff
@@ -32,20 +32,25 @@ namespace {
 
 // Writes the PCD header claiming 'num_points' will follow it into
 // 'output_file'.
-void WriteBinaryPcdHeader(const bool has_color, const int64 num_points,
-                          FileWriter* const file_writer) {
+void WriteBinaryPcdHeader(const bool has_color, const bool has_intensities,
+                          const int64 num_points, FileWriter* const file_writer) {
   std::string color_header_field = !has_color ? "" : " rgb";
   std::string color_header_type = !has_color ? "" : " U";
   std::string color_header_size = !has_color ? "" : " 4";
   std::string color_header_count = !has_color ? "" : " 1";
 
+  std::string intensity_header_field = !has_intensities ? "" : " intensity";
+  std::string intensity_header_type = !has_intensities ? "" : " F";
+  std::string intensity_header_size = !has_intensities ? "" : " 4";
+  std::string intensity_header_count = !has_intensities ? "" : " 1";
+
   std::ostringstream stream;
   stream << "# generated by Cartographer\n"
          << "VERSION .7\n"
-         << "FIELDS x y z" << color_header_field << "\n"
-         << "SIZE 4 4 4" << color_header_size << "\n"
-         << "TYPE F F F" << color_header_type << "\n"
-         << "COUNT 1 1 1" << color_header_count << "\n"
+         << "FIELDS x y z" << intensity_header_field << color_header_field <<"\n"
```

Comment:
Any reason why the ordering between color and intensity is swapped here compared to above for arguments? Some consistency would be nice.

Reply:
The order is not important.

Reply:
You mean not important for correctness, but this is setting the bar too low. Please have a look at https://google.github.io/styleguide/cppguide.html#Goals. Inconsistencies should be there for a reason, not consistency. If you aim for readable code everyone who later reads the code, including yourself, has an easier time.